mreg_cli/api/models.py

```diff
@@ -126,6 +131,38 @@ def parse(cls, value: Any, mode: IPNetMode | None = None) -> IP_AddressT | IP_Ne
             return func(ipnet)
         return ipnet.ip_or_network
 
+    @classmethod
+    def parse_ip_optional(
+        cls, ip: str, version: Literal[4, 6] | None = None
+    ) -> IP_AddressT | None:
```

Comment:
This really shouldn't take a `version` argument, but should instead have a bunch of overloads like `parse` has. However, I got lazy. Will go back and fix.

Reply:
Also, this serves a lot of the same purpose as `IPAddressField.is_valid_ip`, except it returns the IP address as well. They are used in much the same way, except in some places we want to keep the resulting IP (if it exists). 

We should look into unifying this into a single method. It wouldn't even break the current conditionals i.e.:

```
- if IPAddressField.is_valid_ip(ip):
+ if NetworkOrIP.parse_ip_optional(ip):
```

Reply:
Are our semantics closer to `parse` vs `parse_or_raise`?

Reply:
Yes, we should rename it and refactor code that uses what is currently called `parse`.